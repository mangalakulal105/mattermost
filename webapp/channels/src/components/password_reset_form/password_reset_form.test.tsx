// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {fireEvent, render, act, waitFor} from '@testing-library/react';
import {shallow} from 'enzyme';
import React from 'react';

import {resetUserPassword} from 'mattermost-redux/actions/users';

import {withIntl} from 'tests/helpers/intl-test-helper';

import PasswordResetForm from './password_reset_form';

const mockDispatch = jest.fn().mockResolvedValue(Promise.resolve({data: true}));
const mockLocation = jest.fn();
let mockState: any;

jest.mock('mattermost-redux/actions/users', () => ({
    resetUserPassword: jest.fn().mockResolvedValue(Promise.resolve({data: true})),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as typeof import('react-router-dom'),
    useLocation: () => mockLocation(),
    useHistory: () => ({
        push: jest.fn(),
    }),
}));

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux') as typeof import('react-redux'),
    useSelector: (selector: (state: typeof mockState) => unknown) => selector(mockState),
    useDispatch: () => mockDispatch,
}));

describe('components/PasswordResetForm', () => {
    mockState = {
        entities: {
            general: {
                config: {
                    SiteName: 'Mattermost',
                },
            },
        },
    };

    it('should match snapshot', () => {
        mockLocation.mockReturnValue({search: '?token='});
        const wrapper = shallow(<PasswordResetForm/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should call the resetUserPassword() action on submit', async () => {
        mockLocation.mockReturnValue({search: '?token=TOKEN'});

        const wrapper = render(withIntl(<PasswordResetForm/>));
        const inputElement = wrapper.getByTestId('resetPasswordInput');

        act(() => {
            fireEvent.change(inputElement, {target: {value: 'PASSWORD'}});
        });

        await act(async () => {
            fireEvent.click(wrapper.getByRole('button'));
            await waitFor(() => expect(resetUserPassword).toHaveBeenCalledWith('TOKEN', 'PASSWORD'));
        });
    });
});
