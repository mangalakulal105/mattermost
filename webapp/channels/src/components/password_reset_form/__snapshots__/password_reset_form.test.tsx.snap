// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PasswordResetForm should match snapshot 1`] = `
<div
  className="col-sm-12"
>
  <div
    className="signup-team__container reset-password"
  >
    <WomanWithLockSvg />
    <MemoizedFormattedMessage
      defaultMessage="Create a new password"
      id="password_form.title"
      tagName="h1"
    />
    <form
      onSubmit={[Function]}
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Enter a new password for your {siteName} account below."
          id="password_form.enter"
          values={
            Object {
              "siteName": "Mattermost",
            }
          }
        />
        <br />
        Must be 5-72 characters long.
      </p>
      <div
        className="input-line"
      >
        <div
          className="form-group"
        >
          <BrandedInput>
            <ForwardRef
              autoFocus={true}
              className="form-control"
              data-testid="resetPasswordInput"
              id="resetPasswordInput"
              inputSize="large"
              name="password"
              onChange={[Function]}
              placeholder="New password"
              spellCheck="false"
              type="password"
              value=""
            />
          </BrandedInput>
        </div>
        <BrandedButton>
          <button
            className="btn btn-primary"
            id="resetPasswordButton"
            type="submit"
          >
            <MemoizedFormattedMessage
              defaultMessage="Save password"
              id="password_form.change"
            />
          </button>
        </BrandedButton>
      </div>
    </form>
  </div>
</div>
`;
